pkg/utils/sharedticker.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2022 CloudWeGo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import (
+	"sync"
+	"time"
+)
+
+type RefreshTask interface {
+	Refresh()
+}
+
+func NewSharedTicker(refreshInterval time.Duration) *SharedTicker {
+	return &SharedTicker{
+		Interval: refreshInterval,
```

Comment:
1. is it needed to export the field?
2. for all exported funcs, structs you should add comments

Reply:
done